runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -2366,71 +2366,82 @@ void jitClassesRedefined(J9VMThread * currentThread, UDATA classCount, J9JITRede
-            rat->notifyClassRedefinitionEvent(fe, 0,
-               oldMethod,
-               freshMethod);
-
-            // Same as TR_ResolvedMethod::virtualMethodIsOverridden without allocating Scratch Memory
-            if ((UDATA)oldMethod->constantPool & J9_STARTPC_METHOD_IS_OVERRIDDEN)
+            staleMethod = methodList[j].oldMethod;
+            freshMethod = methodList[j].newMethod;
+            bool isSMP = 1; // conservative
+            if (table)
                {
-               UDATA *cp = (UDATA*)&(newMethod->constantPool);
-               *cp |= J9_STARTPC_METHOD_IS_OVERRIDDEN;
+               reportHookDetail(currentThread, "jitClassesRedefined", "    Notify CHTable on method old=%p fresh=%p", oldMethod, freshMethod);
+               table->methodGotOverridden(fe, compInfo->persistentMemory(), (TR_OpaqueMethodBlock*)freshMethod, (TR_OpaqueMethodBlock*)oldMethod, isSMP);
+               }
+            // Step 4 patch modified J9Method
+            if (oldMethod && newMethod && rat)
+               {
+               reportHookDetail(currentThread, "jitClassesRedefined", "    Notify RAT on method old=%p fresh=%p", oldMethod, freshMethod);
+               rat->notifyClassRedefinitionEvent(fe, 0,
+                  oldMethod,
+                  freshMethod);
+
+               // Same as TR_ResolvedMethod::virtualMethodIsOverridden without allocating Scratch Memory
+               if ((UDATA)oldMethod->constantPool & J9_STARTPC_METHOD_IS_OVERRIDDEN)
+                  {
+                  UDATA *cp = (UDATA*)&(newMethod->constantPool);
+                  *cp |= J9_STARTPC_METHOD_IS_OVERRIDDEN;
+                  }
                }
             }
+         // Step 5 patch and update virtual guard. may also need to refresh CHTable
+         if (table)
+            {
+            reportHookDetail(currentThread, "jitClassesRedefined", "  Notify CHTable on class old=%p fresh=%p", oldClass, freshClass);
+            table->classGotRedefined(fe, oldClass, freshClass);
+            }
+
+         classPair = (J9JITRedefinedClass *) ((char *) classPair->methodList + (classPair->methodCount * sizeof(struct J9JITMethodEquivalence)));
          }
-      // Step 5 patch and update virtual guard. may also need to refresh CHTable
-      if (table)
-         {
-         reportHookDetail(currentThread, "jitClassesRedefined", "  Notify CHTable on class old=%p fresh=%p", oldClass, freshClass);
-         table->classGotRedefined(fe, oldClass, freshClass);
-         }
-      classPair = (J9JITRedefinedClass *) ((char *) classPair->methodList + (classPair->methodCount * sizeof(struct J9JITMethodEquivalence)));
       }
+#if defined(J9VM_OPT_JITSERVER)
+   else if (compInfo->getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT)
+      {
+      compInfo->getUnloadedClassesTempList()->push_back(reinterpret_cast<TR_OpaqueClassBlock *>(~0));
+      }
+#endif
```

Comment:
Below, I see some old code that deals with the extended case, but the test is different
```
  if (classCount == 0 || classList == NULL)
    TR::Options::getCmdLineOptions()->setOption(TR_MimicInterpreterFrameShape);
```
Do we have to send that option to the server? It's a global option, so it may not be straightforward.

Reply:
To answer my own question, I think we are fine, because this option can be per compilation. We set it in the global options object, but from there it is copied into the per-compilation options object (in the constructor) when the compilation starts

Reply:
Wouldn't the option be sent in the next compilation request?